hs/src/LedgerState.hs

```diff
@@ -91,8 +90,10 @@ data ValidationError =
   | FeeTooSmall Coin Coin
   -- | Value is not conserved
   | ValueNotConserved Coin Coin
+  -- | Unknown reward account
+  | IncorrectRewards
   -- | The transaction does not have the required witnesses.
-  | InsufficientWitnesses
+  | IncorrectWitnesses
```

Comment:
I think it makes sense to have `insufficient` and `redundant` to be more precise

Reply:
sounds good. I think separating this rule would be easy enough.

Reply:
done.